### PR TITLE
Add new interaction types

### PR DESCRIPTION
**Pull request by hansl**
These allow for constructing a menu that can move pages or handle more cases.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -38,6 +39,30 @@ use embedded_text::{
 
 pub use embedded_menu_macros::{Menu, SelectValue};
 
+/// Internal function to change the selection based on interaction. Moved
+/// separated to allow for easier testing.
+fn calculate_selection(selected: usize, count: usize, interaction: InteractionType) -> usize {
```
Comment (bugadani) [inline src/lib.rs:44]: Hmm, I think this could maybe be a `pub(crate)` method on InteractionType, but never mind that :)

**Review by bugadani — APPROVED** [review]
Thanks a lot, this looks nice :)

File: src/lib.rs (hunk 1 above)
Comment (hansl) [reply src/lib.rs:44]: I can still do that. I have one more change before I'd like a release.

Comment (bugadani) [reply src/lib.rs:44]: I'll leave it up to you, I'm fine with it as it is :)

**Pull request merged**